### PR TITLE
BugFix: Python3 execute file

### PR DESCRIPTION
**Pull request by annakovale**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,7 +1,15 @@
...
-execfile('hooked/version.py')
+
+if sys.version_info[0] < 3:
+    # populate `__version__`
+    execfile('hooked/version.py')
+else:
+    version_file = open("hooked/version.py")
+    exec(version_file.read())
+    version_file.close()
+
```
Comment (marz619) [inline setup.py]: I suggest using `version_info.major` and `with`

```python
_version_file = 'hooked/version.py'

if sys.version_info.major < 3:
    execfile(_version_file)
else:
    with open(_version_file) as vf:
        exec(vf.read())
```

**Comment by bartek** [comment]
I don't understand why we need to use execfile versus just importing the variable from version.py. I must be missing something?

**Comment by annakovale** [comment]
@bartek We probably could, I am just fixing the fact that `execfile` doesn't exist in Python 3 👀

**Comment by marz619** [comment]
@bartek It's the chicken vs egg problem.

If you tried to do the following in the `setup.py` file. It would result in an import module error.
```python
from hooked.version import __version__
``` 

And we don't want to define the `version` number in multiple places, so we do this instead and have the `if...else...` in order to satisfy `py2` and py3+`

**Comment by bartek** [comment]
> @bartek We probably could, I am just fixing the fact that execfile doesn't exist in Python 3 👀

Got it, and thanks @marz619 for additional context. Carry on then.

**Comment by jonprindiville** [comment]
Anna, can you check if the py3 case would _also_ work for py2?

Would be nice to avoid the `if` if we can.

Otherwise, I'll just echo the suggestion for using `with open(...) as foo:` -- this ensures that your file gets closed even if exceptions happen inside of the indented block (saves us from writing a `try/except`, wheeeee)

________________________________

**Pull request merged**